packages/transaction-controller/src/utils/validation.test.ts

```diff
@@ -362,6 +362,60 @@ describe('validation', () => {
           } as any),
         ).not.toThrow();
       });
+
+      it('throws if gasLimit is defined but not a valid hexadecimal', () => {
+        expect(() =>
+          validateTxParams({
+            from: '0x1678a085c290ebd122dc42cba69373b5953b831d',
+            to: '0xfbb5595c18ca76bab52d66188e4ca50c7d95f77a',
+            maxFeePerGas: '0x01',
+            gasLimit: 'zzzzz',
+            // TODO: Replace `any` with type
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          } as any),
```

Comment:
I know these `any`s all over this file but does it make sense to remove all these todo's and mark them as `TransactionParams`

```suggestion
          } as unknown as TransactionParams),
```